### PR TITLE
CRISPR Studio wrapper

### PR DESCRIPTION
**Pull request by KyleLamoureux**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

Wrapper for the new CRISPR_Studio recipe

### CODE REVIEW

File: tools/crispr_studio/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+- Sequence Analysis
+description: CRISPR Studio is a program developed to facilitate and accelerate CRISPR array visualization.
+homepage_url: https://github.com/moineaulab/CRISPRStudio
+long_description: |
+  CRISPR Studio generates a figure comparing the different CRISPR
+  arrays identified by CRISPRDetect. Required File GFF3 file with the CRISPR
+  arrays identified with CRISPRDetect. It works by first comparing spacers sequence 
+  homology in a dataset, then assigning a two-color-code to each cluster of spacers 
+  and finally writing an svg file, which can be opened in graphics vector editor.
+name: CRISPR Studio
```
Comment (bgruening) [inline tools/crispr_studio/.shed.yml]: This should be the toolshed repo name, so I guess `crispr_studio`.

File: tools/crispr_studio/crispr_studio.xml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+        <param name="gray_unique" type="boolean" label="Gray out unique?" help="The unique spacers will be grayed"/>
+
+        <param name="gray_similar" type="boolean" label="Gray out similar?" help="The conserved spacers will be grayed"/>
+
+        <param name="rerun" type="boolean" label="Don't change spacer colour?" help="Use this option to keep the same color attributed to the spacer during a previous analysis"/>
+
+    </inputs>
+    <outputs>
+        <data name="out1" format="tabular" label="${tool.name} on ${on_string}: fasta output" from_work_dir="in.fasta"/>
```
[the remark this answers is not in the document]
Comment (KyleL1998) [reply tools/crispr_studio/crispr_studio.xml]: I added a button which will output every file. Decided to do this I'm pretty sure most of the files are just intermediate steps created during run time, So I just don't show them by default

File: tools/crispr_studio/crispr_studio.xml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            <output name="out1" file="crispr_test.gff.fasta"/>
+            <output name="out2">
+                <assert_contents>
+                    <has_text text="NC_010473"/>
+                </assert_contents>
+            </output>
+            <output name="out3" file="crispr_test.gff.fasta_fasta36.spacermatch"/>
+            <output name="out4" file="crispr_test.gff.fasta_fasta36.spacermatch.mcl"/>
+            <output name="out5" file="crispr_test.gff.fasta_fasta36.spacermatch.mcl.col"/>
+	    <output name="out6" file="crispr_test.gff.fasta_fasta36.spacermatch.mcl.final">
```
[the remark this answers is not in the document]
Comment (KyleL1998) [reply tools/crispr_studio/crispr_studio.xml]: Hopefully it's fixed now. All though I needed to rework that area as I realized I missed a glaring bug.

File: tools/crispr_studio/crispr_studio.xml [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+        <section name="adv" title="Advanced output customizations" expanded="false">
+            <param format="txt" label="List of Isolates to show" name="isolate_list" type="data" optional="true" help="Generate the figure with the subset of isolates listed in the file"/>
+
+            <param format="txt" label="The order of the isolates in the figure" name="sort" type="data" optional="true" help="The order of the isolates in the figure: Available option: CRISPRDetect, DistMatrix or File providing a list of the isolates in the desired order. CRISPRDetect = order in the gff file. DistMatrix = Order extracted from a distance matrix based on the similarity of the arrays in the isolates. File = Order privided in a single column file with the isolates names as the should appear in the figure. The option list of Isolates to show will override this option (Default: DistMatrix)"/>
+
+            <param name="spacer_size" type="boolean" label="Show the size of the spacers over the boxed diamonds?" help="We recommend to use this option only for experimental analysis of the data"/>
```
[the remark this answers is not in the document]
Comment (KyleL1998) [reply tools/crispr_studio/crispr_studio.xml]: changes made.

File: tools/crispr_studio/crispr_studio.xml [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+                    <has_text text="NC_010473"/>
+                </assert_contents>
+            </output>
+            <output name="out7">
+                <assert_contents>
+                    <has_text text="svg"/>
+                </assert_contents>
+            </output>
+        </test>
+    </tests>
```
[the remark this answers is not in the document]
Comment (KyleL1998) [reply tools/crispr_studio/crispr_studio.xml]: Added one.

File: tools/crispr_studio/crispr_studio.xml [hunk 6]
```diff
@@ -0,0 +1,116 @@
+<?xml version="1.0"?>
+<tool id="crispr_studio" name="CRISPR Studio" version="1+galaxy0">
+    <description>
+        facilitate and accelerate CRISPR array visualization
```
[the remark this answers is not in the document]
Comment (KyleL1998) [reply tools/crispr_studio/crispr_studio.xml]: It appears to be bound made the change.

**Comment by KyleL1998** [comment]
I noticed the skip verification option (-f) does not work as the way the author wrote CRISPR Studio he wants the user to hit enter during run time with that flag. I tried `printf '\n' |` but it didn't work on galaxy so I'm going to make a patch to change how this works. So until then this flag will be broken.

**Comment by KyleL1998** [comment]
Made the Change to the recipe

File: tools/crispr_studio/crispr_studio.xml [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        #end if
+
+        -i ./infile
+    ]]></command>
+    <inputs>
+        <param format="gff" label="Input sequences" name="input_data" type="data" help="gff (GFF3) files are valid"/>
+
+        <section name="adv" title="Advanced output customizations" expanded="false">
+            <param name="isolate_list" type="data" format="txt" optional="true" label="List of Isolates to show" help="Generate the figure with the subset of isolates listed in the file"/>
+            <param argument="-s" type="data" format="txt" optional="true" label="The order of the isolates in the figure" help="The order of the isolates in the figure: Available option: CRISPRDetect, DistMatrix or File providing a list of the isolates in the desired order. CRISPRDetect = order in the gff file. DistMatrix = Order extracted from a distance matrix based on the similarity of the arrays in the isolates. File = Order privided in a single column file with the isolates names as the should appear in the figure. The option list of Isolates to show will override this option (Default: DistMatrix)"/>
```
Comment (bgruening) [inline tools/crispr_studio/crispr_studio.xml]: Any reason this is a file input and not a select parameter?

File: tools/crispr_studio/crispr_studio.xml [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+                    <has_text text="NC_010473"/>
+                </assert_contents>
+            </output>
+            <output name="out7">
+                <assert_contents>
+                    <has_text text="svg"/>
+                </assert_contents>
+            </output>
+        </test>
+        <test>
```
Comment (bgruening) [inline tools/crispr_studio/crispr_studio.xml]: a cool feature to thest the number of outputs and test the filtering of outputs is exected_num_outputs, see here: https://docs.galaxyproject.org/en/latest/dev/schema.html#id77

File: tools/crispr_studio/crispr_studio.xml (hunk 7 above)
Comment (KyleL1998) [reply tools/crispr_studio/crispr_studio.xml]: I have no idea what I'd be selecting between. Also the help message for CRISPR Studio makes it seem like it wants input files.

File: tools/crispr_studio/crispr_studio.xml (hunk 8 above)
Comment (KyleL1998) [reply tools/crispr_studio/crispr_studio.xml]: I like it. Added.

**Comment by KyleL1998** [comment]
Hey @bgruening my coop term ends Friday and was hoping this could get merged in as it will get used as well as its one of the last things I have left unfinished.

File: tools/crispr_studio/crispr_studio.xml (hunk 7 above)
Comment (Takadonet) [reply tools/crispr_studio/crispr_studio.xml]: It is a file input since user can provide order of isolate to report.

**Comment by bgruening** [comment]
@Takadonet I added a few smaller changes and gave variables better names. This is good if you want to run the tool via bioblend.

If you could have a look at it again and also at the UX of this tool. Thanks!

**Comment by Takadonet** [comment]
@bgruening please close this PR and review https://github.com/galaxyproject/tools-iuc/pull/2532 . Made a new PR since this one has lots of issue with unrelated tests.

**Pull request closed without merging**